python/lsst/meas/extensions/ngmix/LMSimpleShape.py

```diff
@@ -0,0 +1,238 @@
+    xtol = lsst.pex.config.Field(dtype=float, default=1e-5, optional=False,
+                                 doc="tolerance in solution")
+
+
+@register("meas_extensions_ngmix_LMSimpleShape")
+@lsst.meas.base.flagDecorator.addFlagHandler(("flag", "General Failure error"),
+        ("flag_noPsf", "No PSF attached to the exposure."),
+        ("flag_maxFev", "Maximum function evaluations exceeded"),
+        ("flag_singularMatrix", "Singular Matrix"),
+        ("flag_negativeCovEig", "Negative Covariance Eig"),
+        ("flag_negativeCoveDiag", "Negative Covariance Diag"),
+        ("flag_eigNotFinite", "Eig not finite"),
+        ("flag_funcNotFinite", "Func not finite"),
+        ("flag_divideByZero", "Divide by Zero"))
+class SingleFrameNgmixPlugin(SingleFramePlugin):
+    '''
+    Plugin to do shape measurement using ngmix.LMSimple fitter
+    May be run using gauss, dev, or exp model
+    '''
+    ConfigClass = SingleFrameNgmixConfig
+
+    #   Length of a single gaussian representation in ngmix
+    gaussian_pars_len = 6
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return 3.0
+
+    #   This defines the legal LMSimple routines to call
+    #   This metadata will be fetched from Erin when he adds an API for it
+    def getModelInfo(self, model):
+        modelInfo = {"gauss": {"nPars": 6, "nGauss": 1},
+                     "dev": {"nPars": 6, "nGauss": 6},
+                     "exp": {"nPars": 6, "nGauss": 6}}
+        if model not in modelInfo.keys():
+            raise lsst.pex.exceptions.RuntimeError("Illegal model: %s for LMSimple"%model)
+        return modelInfo[model]
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+
+        # Add ShapeletFunction keys for the number of Gaussians required
+        self.keys = []
+        self.nGauss = self.getModelInfo(self.config.model)["nGauss"]
+        #   Add a 0th order shapelet for every Gaussian to be returned
+        for i in range(self.nGauss):
+            key = ShapeletFunctionKey.addFields(schema, "%s_%d"%(name, i), "ngmix Shape gaussian",
+                                                "pixels", "", 0, HERMITE)
+            self.keys.append(key)
+        #   If the psfName is specified, it means to we can get the psfApprox from it
+        self.msfKey = MultiShapeletFunctionKey(self.keys)
+        if not self.config.psfName is None:
+            self.psfMsfKey = MultiShapeletFunctionKey(schema[self.config.psfName], HERMITE)
+        else:
+            self.psfMsfKey = None
+        self.name = name
+
+    def measure(self, measRecord, exposure):
+        psf = exposure.getPsf()
+        if psf is None:
+            raise MeasurementError(self.flagHandler.getDefinition(
+                                   SingleFrameNgmixPlugin.ErrEnum.flag_noPsf).doc,
+                                   SingleFrameNgmixPlugin.ErrEnum.flag_noPsf)
```

Comment:
As a note, you can either use SingleFrameNgmixPlugin or self here, either will work. If you want to make this shorter with self you can, or if you want to make it clearer that it is a class variable leaving it is fine.